contracts/EthereumDIDRegistry.sol

```diff
@@ -65,26 +70,30 @@ contract EthereumDIDRegistry {
   }
 
   function changeOwnerSigned(address identity, uint8 sigV, bytes32 sigR, bytes32 sigS, address newOwner) public {
-    bytes32 hash = keccak256(byte(0x19), byte(0), this, nonce[identityOwner(identity)], identity, "changeOwner", newOwner); 
+    bytes32 hash = keccak256(byte(0x19), byte(0), this, nonce[identityOwner(identity)], identity, "changeOwner", newOwner);
     changeOwner(identity, checkSignature(identity, sigV, sigR, sigS, hash), newOwner);
   }
 
-  function addDelegate(address identity, address actor, bytes32 delegateType, address delegate, uint validity ) internal onlyOwner(identity, actor) {
+  function addDelegate(address identity, address actor, bytes32 delegateType, address delegate, uint validity, bool _unrevokable) internal onlyOwner(identity, actor) {
     delegates[identity][keccak256(delegateType)][delegate] = block.timestamp + validity;
+    if (_unrevokable) {
```

Comment:
I think for safety sake this should be set to false if _unrevokable is false. Otherwise adding the same key twice would inherit a previously set unrevokable status

@oed

Reply:
That's the point. If you can just set the flag to false again then there is no point of having this.

Reply:
I do `addDelegate(0x1234, true)` to create an unrevokable.

I wait for timeout.

If I now do `addDelegate(0x1234, false)` then that is still going to be unrevokable